### ISSUE TITLE
Falling blocks stay immortals

### ISSUE THREAD

**Issue by Treczoks** [minetest/minetest#14666]
### Minetest version

```true
Minetest 5.9.0-dev (Linux)
Using LuaJIT 2.1.0-beta3
Built by GCC 13.2
Running on Linux/6.5.0-35-generic x86_64
BUILD_TYPE=Release
RUN_IN_PLACE=0
USE_CURL=1
USE_GETTEXT=1
USE_SOUND=1
STATIC_SHAREDIR="/usr/share/minetest"
STATIC_LOCALEDIR="/usr/share/locale"
```

### Irrlicht device

SDL

### Operating system and version

Kubuntu 23.10

### CPU model

4 × Intel® Core™ i3-7020U CPU @ 2.30GHz

### GPU model

Mesa Intel® HD Graphics 620

### Active renderer

OpenGL 4.6

### Summary

Falling blocks turn "immortal", but don't turn back to normal blocks.

### Steps to reproduce

Stack three sand blocks, remove the middle one, upper sand block falls down, turns into an immortal and cannot be handled anymore. It can still fall if you dig out the bottom sand block.

**Comment by Treczoks** [comment]
Forgot to mention: Must be a change in the daily build, as this still worked yesterday.

**Comment by grorp** [comment]
Oddly enough, I accidentally reproduced this bug with MTG while testing #14542 on Windows 10.

**Comment by grorp** [comment]
First bad commit is c24a04d246221e13a626f4a4dd34037a5332a935 @sfan5

**Comment by sfan5** [comment]
```diff
diff --git a/builtin/game/misc_s.lua b/builtin/game/misc_s.lua
--- a/builtin/game/misc_s.lua
+++ b/builtin/game/misc_s.lua
@@ -110,7 +110,7 @@ if core.set_push_moveresult1 then
                        standing_on_object = b2,
                        collisions = {{
                                type = "node",
-                               axis = AXES[axis],
+                               axis = AXES[axis + 1],
                                node_pos = vector.new(npx, npy, npz),
                                new_pos = vector.new(v0x, v0y, v0z),
                                old_velocity = vector.new(v1x, v1y, v1z),
```
probably

**Comment by KaylebJay** [comment]
Tested sfan5's patch, works!

**Comment by Treczoks** [comment]
Nice to see that this got addressed so quickly. Will the fix be in a daily?

Now how will I get rid of those immortal stacks of sand and gravel? They are located rather inconveniently, and I have not found a way to "kill" them.

**Comment by grorp** [comment]
Fix merged: c4703a7f19441b2262630a34d64f9f1c210a9375

> Now how will I get rid of those immortal stacks of sand and gravel? 

They will disappear by themselves.
